Simulator keyboard navigation

glastonbridge opened the pull request:
Provides the following accessibility improvements in the micro:bit simulator:
* re-structures the DOM so that tab ordering of simulator elements is left-right top-bottom (see videos)
* makes the antenna element interactive using keyboard navigation, simplest approach
    * previously it was an invisible tab stop
* add some preventDefaults to prevent the screen scrolling while you're adjusting values when the simulator has a scrollbar

## Old behaviour

Sequential tabbing through the simulator using the tab key

https://github.com/user-attachments/assets/130633e1-2751-45ea-b9b2-0e5179cfb8b6

## New behaviour
https://github.com/user-attachments/assets/5238ff6b-76bb-4ff0-87cd-fccb321d7157

[You can try it here](https://simulator-keyboard-navigatio.pxt-microbit.pages.dev/).

[Example project](https://makecode.microbit.org/_0qwJ3JKjzerk) (needs JavaScript copy/pasting to correct deployment).

microbit-matt-hillsdon commented:
Deployment link: https://simulator-keyboard-navigatio.pxt-microbit.pages.dev/

microbit-matt-hillsdon commented on sim/visuals/microbit.ts:
```diff
@@ -966,7 +976,7 @@ path.sim-board {
         private buildDom() {
             this.domHardwareVersion = 1;
-            this.element = <SVGSVGElement>svg.elt("svg")
+            this.element = <SVGSVGElement>svg.elt("svg");
```
There are a few lines like this where we've added semis that distract from this PR's focus. Please can we remove to keep a simple diff?

glastonbridge committed:
Accessibility improvements to the micro:bit simulator.

- re-structures the DOM so that tab ordering of simulator elements moves
  left-to-right and top-to-bottom.
- Makes the antenna element interactive using keyboard navigation, simple
  approach. Previously an invisible tab stop with no keyboard interaction.
- Add preventDefault to keyboard actions to prevent scrolling when setting
  values when the simulator is inside a scrolled container.
